scipy/optimize/tests/test_optimize.py

```diff
@@ -1002,6 +1003,7 @@ def dfunc(z):
             assert_(func(sol1.x) < func(sol2.x),
                     "%s: %s vs. %s" % (method, func(sol1.x), func(sol2.x)))
 
+    @pytest.mark.filterwarnings('ignore::UserWarning')
```

Comment:
If the test always generates a warning, this should be replaced with `with pytest.warns(OptimizeWarning):`.